geoopt/manifolds/product.py

```diff
@@ -435,3 +437,96 @@ def origin(
             )
         tensor = self.pack_point(*points)
         return geoopt.ManifoldTensor(tensor, manifold=self)
+
+
+class StereographicProductManifold(ProductManifold):
+    """
+    Product Manifold for Stereographic manifolds.
+
+    Examples
+    --------
+    A Torus
+
+    >>> import geoopt
+    >>> sphere = geoopt.SphereProjection()
+    >>> torus = StereographicProductManifold((sphere, 2), (sphere, 2))
+    """
+
+    def __init__(
+        self,
+        *manifolds_with_shape: Tuple[Stereographic, Union[Tuple[int, ...], int]],
+    ):
+        super().__init__(*manifolds_with_shape)
```

Comment:
I'd add a check after the parent init to verify all manifolds are the same family

Reply:
```suggestion
        super().__init__(*manifolds_with_shape)
        for man in self.manifilds:
            if not geoopt.utils.ismanifold(man, Stereographic):
                raise TypeError("Every submanifold as to be Stereographic manifold")
```